lightkurve/search.py

```diff
@@ -1023,7 +1023,7 @@ def open(path_or_url, **kwargs):
     # pass header into `detect_filetype()`
     try:
         with fits.open(path_or_url) as temp:
-            filetype = detect_filetype(temp[0].header)
+            filetype = detect_filetype(temp[0].get_header())
```

Comment:
The tests are failing because `temp[0]` is a FITS HDU object here, i.e. the use of `.header` here is correct and not the cause of the warnings.